service/reducers/entities/files.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {FilesTypes, UsersTypes} from 'service/constants';
+
+function files(state = {}, action) {
+    switch (action.type) {
+    case FilesTypes.RECEIVED_FILES_FOR_POST: {
+        const filesById = action.data.reduce((filesMap, file) => {
+            return {...filesMap,
+                [file.id]: file
+            };
+        }, {});
+        return {...state,
+            ...filesById
+        };
+    }
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+    default:
+        return state;
+    }
+}
+
+function fileIdsByPostId(state = {}, action) {
```

Comment:
@hmhealey That was my first approach, to just use `post.file_ids` to select the associated files, but then I decided that what the server returns for `get_file_infos` is the source of truth. It's conceivable that a post could still have an id for a file that no longer exists on the server.
But now I'm realizing that if we allow for adding/removing an existing post's file attachments, then it'll be simpler to handle that update if we're not storing a separate map.